cluster.go

```diff
@@ -281,6 +293,34 @@ func (c *Cluster) Sync() error {
 	return err
 }
 
+func isTopoContains(topo ClusterTopo, node ClusterNode) bool {
+	for _, t := range topo.Map() {
+		if reflect.DeepEqual(t, node) {
+			return true
+		}
+	}
+	return false
+}
+
+func (c *Cluster) traceTopoChanged(prevTopo ClusterTopo, newTopo ClusterTopo) {
+	if c.co.ct.TopoChanged != nil {
+		var Added []trace.ClusterNodeInfo
+		var Removed []trace.ClusterNodeInfo
+		for _, t := range newTopo.Map() {
```

Comment:
`Map` returns a map keyed by the node address. You could just check which keys are in the old map that are missing in the new map and vice versa. For keys that exists in both you could compare IDs and slots and track changes separate from added and removed nodes.

Reply:
That's a good idea. I'll change it to iterate newTopo's `Map` and check whether it was added or changed.